batch7crud-roll7/src/main/java/com/lftechnology/batch7crud/controller/StudentListController.java

```diff
@@ -0,0 +1,187 @@
+            else
+                showErrorPage(request,response);
+        } catch (DataException | ServletException | IOException e) {
+            LOGGER.log(Level.SEVERE, e.getMessage(), e);
+        }
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String[] urlpath = urlParts(request);
+        try {
+            if (urlpath.length == 2 )
+                list(request, response);
+            else if(urlpath.length == 3 && "NewEntry".equals(urlpath[2]))
+                create(request, response);
+            else if(urlpath.length == 4 && "edit".equals(urlpath[3]))
+                edit(request, response, Integer.parseInt(urlpath[2]));
+            else if(urlpath.length == 3)
+                show(request,response);
+            else
+                showErrorPage(request, response);
+        } catch (DataException | ServletException | IOException e) {
+            LOGGER.log(Level.SEVERE, e.getMessage(), e);
+        }
+    }
+
+    private void list(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException, DataException {
+        StudentServices stdservice = new StudentServices();
+        int pageSize = 3;
+        int page = 1;
+        try{
+            page = getPageNumber(request);
+        }
+        catch(NumberFormatException e){
+            showErrorPage(request, response);
+        }
+        int totalStudents = stdservice.fetchTotal();
+
+        if(page !=1 && page > Math.ceil(totalStudents/(float)pageSize)){
+            showErrorPage(request,response);
+        }
+        request.setAttribute("students", stdservice.fetch(page,pageSize));
+        request.setAttribute("pageSize",pageSize);
+        request.setAttribute("totalStudents",totalStudents);
+        request.setAttribute("pageNum",page);
+
+        request.getServletContext().getRequestDispatcher("/WEB-INF/views/students.jsp").forward(request, response);
+
+    }
+
+    private void create(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        request.getServletContext().getRequestDispatcher("/WEB-INF/views/newEntry.jsp").forward(request, response);
+    }
+
+    private void createProcess(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException, DataException {
+        String name = request.getParameter("name");
+        String address = request.getParameter("address");
+        String roll = request.getParameter("roll");
+        try{
+
+            int rollNum = Integer.parseInt(roll);
+            Student s = new Student();
+            s.setName(name);
+            s.setAddress(address);
+            s.setRoll(rollNum);
+
+            StudentServices stdServices = new StudentServices();
+            stdServices.addNew(s);
+            response.sendRedirect("/Students");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "/Students" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)